app/controllers/v1/nod_callbacks_controller.rb

```diff
@@ -14,20 +14,33 @@ class NodCallbacksController < ApplicationController
     skip_after_action :set_csrf_header, only: [:create]
     before_action :authenticate_header, only: [:create]
 
+    STATUSES_TO_PERSIST = %w[permanent-failure technical-failure preferences-declined].freeze
+
     def create
       return render json: nil, status: :not_found unless Flipper.enabled? :nod_callbacks_endpoint
 
+      payload = JSON.parse(request.body.string)
+
       log_params = {
         key: :callbacks,
         form_id: '10182',
         user_uuid: nil,
         upstream_system: 'VANotify',
-        body: JSON.parse(request.body.string)
+        body: payload.merge('to' => '<FILTERED>') # scrub PII from logs
       }
-      log_formatted(**log_params.merge(is_success: true))
 
-      # TODO: save encrypted request body in database table for non-successful notifications
+      # save encrypted request body in database table for non-successful notifications
+      payload_status = payload['status']&.downcase
+      if STATUSES_TO_PERSIST.include? payload_status
```

Comment:
It may be a good idea to instead check whether payload_status isn't included in the successful statuses. Just in case we get any unexpected statuses from VANotify that turn out to be non-successful statuses.